src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1869,10 +1899,62 @@ private File createSshKeyFile(SSHUserPrivateKey sshUser) throws IOException, Int
                 w.println(s);
             }
         }
-        new FilePath(key).chmod(0400);
+        if (launcher.isUnix()) {
+            new FilePath(key).chmod(0400);
+        } else {
+            fixSshKeyOnWindows(key);
+        }
+
         return key;
     }
 
+    /* package protected for testability */
+    void fixSshKeyOnWindows(File key) throws GitException {
+        if (launcher.isUnix()) return;
+
+        Path file = Paths.get(key.toURI());
+
+        AclFileAttributeView fileAttributeView = Files.getFileAttributeView(file, AclFileAttributeView.class);
+        if (fileAttributeView == null) return;
+
+        String username = getWindowsUserName(fileAttributeView);
+        if (StringUtils.isBlank(username)) return;
+
+        try {
+            UserPrincipalLookupService userPrincipalLookupService = file.getFileSystem().getUserPrincipalLookupService();
+            UserPrincipal userPrincipal = userPrincipalLookupService.lookupPrincipalByName(username);
+            AclEntry aclEntry = AclEntry.newBuilder()
+                .setType(AclEntryType.ALLOW)
+                .setPrincipal(userPrincipal)
+                .setPermissions(ACL_ENTRY_PERMISSIONS)
+                .build();
+            fileAttributeView.setAcl(Collections.singletonList(aclEntry));
+        } catch (IOException | UnsupportedOperationException e) {
+            throw new GitException("Error updating file permission", e);
```

Comment:
Could the exception message include the name of the file that failed the permission setting operation?  

That will help us more quickly diagnose the failure in those cases where the caller shows the exception message rather than showing the entire exception stack trace.

Reply:
will remove the stack trace and print the full file path?

Reply:
@MarkEWaite done 🙌